src/Raven.Server/Documents/AI/Embeddings/IEmbeddingValue.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.IO;
+
+namespace Raven.Server.Documents.AI.Embeddings;
+
+public interface IEmbeddingValue
```

Comment:
What does this gives us that `ReadOnlyMemory<byte>` does not?

Reply:
There are 3 implementations:

- `EmbeddingValue`  - where you also provide `usedBytes` parameter which is used for quantization (as I know it does in place quantization to avoid extra allocations so the type remains `ReadOnlyMemory<float>` but less bytes is used)
- `StreamedEmbeddingValue` - used by attachments, in particular for embeddings that are already cached
- `EmbeddingTestValue` - used in unit test


In general the motivation for adding this was to have better way of dealing with embeddings coming from different sources - embeddings generated by AI service (so ReadOnlyMemory<float>) and cached embeddings (coming from local attachments storage - Stream). The both might be used at the same time during query execution where we need to convert them to Corax type - `VectorValue`.

Reply:
There should be just `ReadOnlyMemory<byte>` for everything, I think.